src/basis/Symbol.mli

```diff
@@ -1,11 +1,20 @@
-type t
-
-val fresh : unit -> t
-val named : string -> t
-val named_opt : string option -> t
+(* FIXME: This is anti-modular, but it makes the symbol
+   table code less janky for now.
+ *)
+type t = { origin : string; index : int; name : string option}
```

Comment:
BTW the anti-modularity worries me a bit. Not all global symbols will have a unit origin.

Reply:
Yeah, I don't love this either... It's the only way I could think of that avoids some of the hairy symbol rebasing stuff however. Out of curiousity, what would be a symbol that doesn't originate in some code unit?

Reply:
One possibility is a unification variable, which may or may not have an origin (depending on the design).